### PR TITLE
Remove unshare_root workaround

### PR DESCRIPTION
**Pull request by marcmo**
by introducing a bind-mount of the run-dir
and marking the mount propagation for that one as private

### CODE REVIEW

**Comment by marcmo** [comment]
Not sure that is all. Running on my machine the mount is not visible to other processes.
tests are now broken but first wanted to get some feedback from @flxo

**Comment by flxo** [comment]
@marcmo The tests fail with:

```
The application panicked (crashed).
956
Message:  Test failed: 
957
   0: Failed to remove tmpdir
958
   1: Device or resource busy (os error 16) at path "/tmp/.tmpOnsgZJ"
```

This is because the bind mount is not umounted upon shutdown. This missing  when the runtime task breaks.

File: northstar/src/main.rs [hunk 1]
```diff
@@ -59,15 +59,6 @@ fn main() -> Result<(), Error> {
         .map(|r| r.disable_mount_namespace)
         .unwrap_or(false)
     {
-        // Set the mount propagation of unshare_root to MS_PRIVATE
```
Comment (flxo) [inline northstar/src/main.rs:62]: This needs to be remove from integration_tests::macros.rs as well

File: northstar_tests/src/runtime.rs [hunk 2]
```diff
@@ -94,7 +94,6 @@ impl Northstar {
             cgroups,
             // This sections matches most common x86 Linux distributions
             devices: config::Devices {
-                unshare_root: PathBuf::from("/"),
```
Comment (flxo) [inline northstar_tests/src/runtime.rs:97]: The `unshare_root` is in the example northstar.toml. Think we should add `#[serde(deny_unknown_fields)]` to not allow unknown fields in all struct in `config.rs`.

**Review by flxo — CHANGES_REQUESTED** [review]

**Pull request merged**